### PR TITLE
New Ignore rule for difftool to ignore the changes in RAML examples

### PR DESCRIPTION
**Pull request by kalagarraj**
Included changes
- New default rule to ignore example changes
- Test cases for default rule and node type structural changes
- Sorted package.json

### CODE REVIEW

File: src/diff/changes/nodeChanges.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ import { RuleCategory } from "../ruleSet";
 export class NodeChanges {
   public added: { [key: string]: unknown };
   public removed: { [key: string]: unknown };
+  public ignored: { [key: string]: unknown };
```
Comment (unandyala) [inline src/diff/changes/nodeChanges.ts:17]: what is the purpose of ignored object? I don't see it being used anywhere

File: src/diff/defaultRules.test.ts [hunk 2]
```diff
@@ -27,7 +31,7 @@ function verifyRule(nodeChanges: NodeChanges, rule: Rule): void {
       nodeChanges.added[rule.event.params.changedProperty]
     ];
   }
-  expect(categorizedChanges).to.have.length(1);
+  expect(categorizedChanges).to.have.length(countOfChanges);
```
Comment (unandyala) [inline src/diff/defaultRules.test.ts:34]: length seems to have deprecated. We should use lengthOf instead

File: src/diff/defaultRules.test.ts [hunk 3]
```diff
@@ -120,3 +124,75 @@ describe("Version change", () => {
     verifyRule(changes[0], rule);
   });
 });
+
+describe("Examples change", () => {
+  it("does not apply ignore examples rule on a change that has no apiContract:Example change type", async () => {
```
Comment (unandyala) [inline src/diff/defaultRules.test.ts:129]: apiContract:Example is a node type not a change type

Comment (kalagarraj) [reply src/diff/defaultRules.test.ts]: Done

File: src/diff/defaultRules.test.ts (hunk 2 above)
Comment (kalagarraj) [reply src/diff/defaultRules.test.ts]: done

File: src/diff/changes/nodeChanges.ts (hunk 1 above)
Comment (kalagarraj) [reply src/diff/changes/nodeChanges.ts]: Removed

File: resources/diff/rules/defaultRules.json [hunk 4]
```diff
@@ -164,5 +164,24 @@
...
+        {
+          "fact": "diff",
+          "path": "$.type",
+          "operator": "contains",
+          "value": "apiContract:Example"
+        }
+      ]
+    },
+    "event": {
+      "type": "example-change",
```
Comment (unandyala) [inline resources/diff/rules/defaultRules.json:181]: change type to "example-changed" - just to be consistent with types in other rules

Comment (kalagarraj) [reply resources/diff/rules/defaultRules.json]: Makes sense, done..

**Pull request merged**